### DISCUSSION TITLE
Trouble with label JSON formatting for text detection training

### DISCUSSION THREAD

**Discussion by dvivarelli** [mindee/doctr#739] [Q&A] [answered]
Hello,
I'm traing to train on my own dataset the detection phase.

I have made some json for training and validation, but during trainig i receive this error:

"{IndexError}index 2 is out of bounds for axis 2 with size 2"  at detction.py line 78  ( boxes[..., [0, 2]] /= w  )

Could you help me  to understand what's wrong?

here an exemple taken from labels.json. The polygons are box so only 4 points

{
  "19d447e2-143refed.jpg": {
    "polygons": [
      [
        [
          [
            206 ,
            95
          ] ,
          [
            340 ,
            95
          ] ,
          [
            340 ,
            106
          ] ,
          [
            206 ,
            106
          ]
        ] ,
        [
          [
            358 ,
            94
          ] ,
          [
            388 ,
            94
          ] ,
          [
            388 ,
            105
          ] ,
          [
            358 ,
            105
          ]
        ] ,
        [
          [
            393 ,
            95
          ] ,
          [
            403 ,
            95
          ] ,
          [
            403 ,
            106
          ] ,
          [
            393 ,
            106
          ]
        ] ,
        [
          [
            234 ,
            40
          ] ,
          [
            335 ,
            40
          ] ,
          [
            335 ,
            51
          ] ,
          [
            234 ,
            51
          ]
        ] ,
        [
          [
            153 ,
            51
          ] ,
          [
            314 ,
            51
          ] ,
          [
            314 ,
            65
          ] ,
          [
            153 ,
            65
          ]
        ] ,
        [
          [
            180 ,
            65
          ] ,
          [
            328 ,
            65
          ] ,
          [
            328 ,
            75
          ] ,
          [
            180 ,
            75
          ]
        ] ,
        [
          [
            218 ,
            122
          ] ,
          [
            282 ,
            122
          ] ,
          [
            282 ,
            133
          ] ,
          [
            218 ,
            133
          ]
        ] ,
        [
          [
            219 ,
            142
          ] ,
          [
            254 ,
            142
          ] ,
          [
            254 ,
            153
          ] ,
          [
            219 ,
            153
          ]
        ] ,
        [
          [
            217 ,
            164
          ] ,
          [
            292 ,
            164
          ] ,
          [
            292 ,
            174
          ] ,
          [
            217 ,
            174
          ]
        ] ,
        [
          [
            160 ,
            141
          ] ,
          [
            191 ,
            141
          ] ,
          [
            191 ,
            152
          ] ,
          [
            160 ,
            152
          ]
        ] ,
        [
          [
            159 ,
            123
          ] ,
          [
            209 ,
            123
          ] ,
          [
            209 ,
            135
          ] ,
          [
            159 ,
            135
          ]
        ] ,
        [
          [
            161 ,
            95
          ] ,
          [
            197 ,
            95
          ] ,
          [
            197 ,
            105
          ] ,
          [
            161 ,
            105
          ]
        ] ,
        [
          [
            159 ,
            158
          ] ,
          [
            193 ,
            158
          ] ,
          [
            193 ,
            171
          ] ,
          [
            159 ,
            171
          ]
        ] ,
        [
          [
            160 ,
            186
          ] ,
          [
            205 ,
            186
          ] ,
          [
            205 ,
            197
          ] ,
          [
            160 ,
            197
          ]
        ] ,
        [
          [
            217 ,
            185
          ] ,
          [
            237 ,
            185
          ] ,
          [
            237 ,
            197
          ] ,
          [
            217 ,
            197
          ]
        ] ,
        [
          [
            200 ,
            204
          ] ,
          [
            284 ,
            204
          ] ,
          [
            284 ,
            216
          ] ,
          [
            200 ,
            216
          ]
        ] ,
        [
          [
            159 ,
            204
          ] ,
          [
            186 ,
            204
          ] ,
          [
            186 ,
            216
          ] ,
          [
            159 ,
            216
          ]
        ] ,
        [
          [
            68 ,
            158
          ] ,
          [
            107 ,
            158
          ] ,
          [
            107 ,
            170
          ] ,
          [
            68 ,
            170
          ]
        ] ,
        [
          [
            78 ,
            186
          ] ,
          [
            152 ,
            186
          ] ,
          [
            152 ,
            198
          ] ,
          [
            78 ,
            198
          ]
        ] ,
        [
          [
            161 ,
            104
          ] ,
          [
            198 ,
            104
          ] ,
          [
            198 ,
            115
          ] ,
          [
            161 ,
            115
          ]
        ] ,
        [
          [
            159 ,
            170
          ] ,
          [
            209 ,
            170
          ] ,
          [
            209 ,
            180
          ] ,
          [
            159 ,
            180
          ]
        ] ,
        [
          [
            68 ,
            170
          ] ,
          [
            116 ,
            170
          ] ,
          [
            116 ,
            181
          ] ,
          [
            68 ,
            181
          ]
        ] ,
        [
          [
            159 ,
            216
          ] ,
          [
            208 ,
            216
          ] ,
          [
            208 ,
            226
          ] ,
          [
            159 ,
            226
          ]
        ]
      ]
    ] ,
    "img_hash": "08deae8d9ea9bc0b84f94475d868351830e9f7e7" ,
    "img_dimensions": "(526,289)"
  }

**Comment by fg-mindee** [comment]
Hi @dvivarelli :wave: 

I must confess I'm a bit puzzled by your question, as I don't understand how you got this. So let's try to solve it:
- could you first specify the exact command you used to run your training?
- could you send the full traceback (not only the last line of the error, everything that shows where it originated from)? (please make sure to put those between triple quotes to format them accordingly)
- could you run the collect_env script and print the result here please? Use the following command:
```shell
wget https://raw.githubusercontent.com/mindee/doctr/main/scripts/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```

**Reply by dvivarelli** [reply]
thanks for your help.
i have found the issue, json was wrong. You can close this ticket

**Reply by fg-mindee** [reply]
Glad you found the issue!
For further reference, would you mind writing a new "answer" (not commenting in this thread, but at the bottom of the page) specifying which was the formatting error in your JSON?

That will help people coming to this page to solve their issue :)

**Reply by dvivarelli** [reply]
There was 3 nasted arrays :
"polygons": [
[  <-- this was wrong
  [
    [
    206 ,
    95
    ] ,
    [
    ....

instead it should be
"polygons": [
  [
    [
    206 ,
    95
    ] ,
    [
     ...

**Comment by fg-mindee** [comment, marked as answer]
For people visiting this page, @dvivarelli found the way to solve this: there was an issue in the json formatting (cf. https://github.com/mindee/doctr/discussions/739#discussioncomment-1867846)

The polygons in the JSON were 3-dimensional while they should be 2-dimensional :+1:
